### ISSUE TITLE
Documentation of what can be in a package_spec

### ISSUE THREAD

**Issue by jmarshall** [conda/conda#10491]
`conda create --help` and `conda install --help` say

```
usage: conda create [options…] [package_spec [package_spec ...]]

Create a new conda environment from a list of specified packages. To use the created
environment, use 'conda activate envname' look in that directory first.  This command
requires either the -n NAME or -p PREFIX option.

Options:

positional arguments:
  package_spec          Packages to install or update in the conda
                        environment.
[etc]
```
```
usage: conda install [options…] [package_spec [package_spec ...]]

Installs a list of packages into a specified conda environment.

    This command accepts a list of package specifications (e.g, bitarray=0.8)
    and installs a set of packages consistent with those specifications and
    compatible with the underlying environment. If full compatibility cannot
    be assured, an error is reported and the environment is not changed.

    Conda attempts to install the newest versions of the requested packages. To
    accomplish this, it may update some packages that are already installed, or
    install additional packages. To prevent existing packages from updating,
    use the --freeze-installed option. This may force conda to install older
    versions of the requested packages, and it does not prevent additional
    dependency packages from being installed.

    If you wish to skip dependency checking altogether, use the '--no-deps'
    option. This may result in an environment with incompatible packages, so
    this option must be used with great caution.

    conda can also be called with a list of explicit conda package filenames
    (e.g. ./lxml-3.2.0-py27_0.tar.bz2). Using conda in this mode implies the
    --no-deps option, and should likewise be used with great caution. Explicit
    filenames and package specifications cannot be mixed in a single command.

Options:

positional arguments:
  package_spec          Packages to install or update in the conda
                        environment.
[etc]
```
 

I did not find any documentation explaining exactly what can appear in a `package_spec`. The [Managing Packages](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#installing-packages) page on docs.conda.io gives examples including `conda install scipy` and `conda install scipy=0.15.0` but I believe other operators (e.g. `scipy>=0.15.0`) are also available.

It would be great if there was a documentation page explaining:

* Exactly what syntax can appear in a `package_spec` on create and install command lines

* What the list of operators available is, and whether there is any difference between `scipy=0.15.0` and `scipy==0.15.0`

* What caveats, if any, there are with using a plain unqualified package_spec as in `conda install scipy`.
 
     The `install --help` appears to say that this will install the latest version of scipy (“Conda attempts to install the newest versions of the requested packages“). Does this mean the newest available in the selected channels, or the newest available whose dependencies are satisfiable from the selected channels, or… perhaps other caveats as well.

**Comment by ickc** [comment]
I find that info in the docstring here: https://github.com/conda/conda/blob/05b3e799daba158b74c892bf3aeb60ffe7485e3c/conda/models/match_spec.py#L72-L153

So the issue is that it is not included in the generated documentation.

**Comment by jmarshall** [comment]
> “Documentation of what can be in a package_spec” … “`conda create --help` and `conda install --help`” … “The Managing Packages page on docs.conda.io” … “would be great if there was a documentation page explaining”.

Yes, this issue is indeed about **user-facing** documentation. Thanks for your input.

**Comment by travishathaway** [comment]
Hi @jmarshall,

Thanks for filing this feature request to improve the documentation. We agree that this would be nice a page to have in our documentation and to help this issue get more attention I've added the https://github.com/conda/conda/labels/good%20first%20issue label.

**Comment by jmarshall** [comment]
@travishathaway: Do you really think a neophyte contributor would consider themselves qualified to write reference documentation for this tool?

Moreover, the wider point of this issue is to document the non-obvious caveats alluded to in the last bullet point above:

* Implications of channel selections — e.g. bioconda documents a need for conda-forge, but if you don't enable conda-forge many `conda install <biocondapkg>` commands will give you an old version of that package that happens not to use anything from conda-forge, instead of failing. This is a common user error.

* Limitations of the _current_repodata.json_ optimisation — cf #9905.

Only experienced conda maintainers have the background to describe these caveats accurately.

**Comment by michaelkarlcoleman** [comment]
@jmarshall : You never know.  Way back in the mysts of time, I wrote the first detailed `ptrace(2)` man page for Linux.  How?  By simply wading through everything in the kernel source that was related.  I didn't really know much about how it worked when I began--I certainly wasn't an experienced kernel maintainer.  But, I did really need it for a project.

Anyway, split the middle?  Certainly a caveat is in order: To do this right, you're going to have to read the source code and figure out exactly how it all works.

**Comment by travishathaway** [comment]
@michaelkarlcoleman,

I still think this would be a great task for a first time contributor. The conda community is very welcoming and would be happy to assist on this during the review process. We also have a chat room where anyone can stop by to ask questions: https://conda.org/community#chat

**Comment by jmarshall** [comment]
The issue still exists unchanged.

**Comment by codewithdaniel1** [comment]
Hi! I’m a student in the CodePath AI301 Open Source program, and I’d like to work on this documentation issue.

I’ll start by checking the current conda docs and CLI help output for `package_spec`, then identify where a concise explanation of valid package spec syntax should live. My plan is to keep the change documentation-only, add examples for common version/operator formats, and make sure the wording matches existing conda terminology.

Please let me know if there is a preferred documentation page or style pattern I should follow.

**Comment by michaelkarlcoleman** [comment]
@codewithdaniel1 I'm not associated with the project, aside from having been a user, but thank you for taking this on!

**Comment by danyeaw** [comment]
Thanks @jmarshall for opening this issue and for your patience with a fix. Kudos to @codewithdaniel1 who updated the docs to make things much more clear.

**Comment by jmarshall** [comment]
To be clear, the pull request does not attempt to describe the caveats that were the reason for opening this issue.

**Comment by danyeaw** [comment]
Hey @jmarshall, I'm not sure I understand what you mean by caveats. The docs now explain the most common package spec items (versions, channels, and build strings), and the complete MatchSpec specification is now linked in CEP 29.

**Comment by jmarshall** [comment]
Re-read [my comment](https://github.com/conda/conda/issues/10491#issuecomment-1569209076) from three years ago.

Channel selection mistakes remain common and lead to ancient packages being installed when the user thinks they're selecting the most up-to-date. With the passage of time and the replacement of various internals, issues due to _current_repodata.json_ (cf #9905, conda/conda-build#4187) may no longer be relevant.

**Comment by danyeaw** [comment]
There is a dedicated docs page about channel handling here: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html

Bioconda has docs about their recommended channel configuration here: https://bioconda.github.io/

If someone is running `conda install scipy` in a fresh environment and not getting the latest version, that sounds like a bug that we should fix.

We can always improve the docs more, but I think we need to better define the caveats that are missing documentation.
